packages/KDateTimePicker/KDateTimePicker.vue

```diff
@@ -0,0 +1,632 @@
+  props: {
+    /**
+     * Provides a default Date object to seed the v-calendar instance
+     */
+    defaultCustom: {
+      type: Object | Number,
+      required: false,
+      default: ''
+    },
+    /**
+     * Default relative time frame object, displayed as pre-selected choice
+     */
+    defaultRelative: {
+      type: Object,
+      required: false,
+      default: () => {}
+    },
+    /**
+     * Upper bound for `v-calendar` dates, everything after this date will be disabled
+     */
+    maxDate: {
+      type: Number,
+      required: false,
+      default: null
+    },
+    /**
+     * Lower bound for `v-calendar` dates, everything preceding this date will be disabled
+     */
+    minDate: {
+      type: Number,
+      required: false,
+      default: null
+    },
+    /**
+     * Determines which `v-calendar` type to initialize.
+     * Three of the values (`date`, `time`, `dateTime`) are passed verbatim to `v-calendar`,
+     * whereas `relative` denotes a component instance made up solely of time frames.
+     */
+    mode: {
+      type: String,
+      required: true,
+      validator: (value) => {
+        return [ 'relative', 'date', 'time', 'dateTime' ].includes(value)
+      }
+    },
+    /**
+     * Help text displayed as the default mesage inside the input field.
+     * When "Clear" is clicked, the input will revert to displaying this.
+     */
+    placeholder: {
+      type: String,
+      required: false,
+      default: 'Select a time range'
+    },
+    /**
+     * Determines whether the `v-calendar` will allow a single date/time,
+     * or a range of dates/times.
+     */
+    range: {
+      type: Boolean,
+      required: false,
+      default: false
+    },
+    /**
+     * A custom set of time frames to be displayed as selectable buttons.
+     * The `timeframeLength`, `start`, and `end` values are passed in as functions,
+     * allowing for on-the-fly date boundary creation.
+     */
+    timePeriods: {
+      type: Array,
+      required: false,
+      default: () => [],
+      validator: (valuesArray) => {
+        return valuesArray.every((item) => {
+          return Array.isArray(item.values) && item.values.every((timeframe) => {
+            // Check validity of each timeframe
+            return typeof timeframe.timeframeText === 'string' &&
+              timeframe.timeframeLength !== undefined &&
+              typeof timeframe.key === 'string' &&
+              timeframe.key !== 'undefined' &&
+              typeof timeframe.prefix === 'string' &&
+              timeframe.prefix !== 'undefined' &&
+              timeframe.start !== undefined &&
+              timeframe.end !== undefined
+          })
+        })
+      }
+    }
+  },
+  setup (props, { emit }) {
+    const hidePopover = ref(false)
+    const selectedRange = ref({ start: '', end: '', timePeriodsKey: '' })
+    const hasDefaultCustomValue = (props.defaultCustom.start && props.defaultCustom.end) || props.defaultCustom !== ''
```

Comment:
Can we make this a `ref` like all the others?

Reply:
Will do.  Maybe for the second pass on this, I can stuff all the state-related things into a massive `reactive` object ( and thereby remove 49 messy instances of `.value` 🤔  )